src/documents/migrations/1053_document_pages_count.py

```diff
@@ -0,0 +1,109 @@
+# Generated by Django 4.2.16 on 2024-09-21 15:44
+
+import datetime
+from pathlib import Path
+
+import pikepdf
+from django.conf import settings
+from django.db import migrations
+from django.db import models
+from django.utils import timezone
+from django.utils.termcolors import colorize as colourise
+
+from documents.parsers import get_default_file_extension
+
+
+class Document:
+    """
+    Django's migrations restrict access to model methods, so this is a snapshot
+    of the methods that existed at the time this migration was written, since
+    we need to make use of a lot of these shortcuts here.
+    """
+
+    def __init__(self, doc):
+        self.pk = doc.pk
+        self.correspondent = doc.correspondent
+        self.title = doc.title
+        self.mime_type = doc.mime_type
+        self.filename = doc.filename
+        self.created = doc.created
+
+    def __str__(self) -> str:
+        # Convert UTC database time to local time
+        created = datetime.date.isoformat(timezone.localdate(self.created))
+
+        res = f"{created}"
+
+        if self.correspondent:
+            res += f" {self.correspondent}"
+        if self.title:
+            res += f" {self.title}"
+        return res
+
+    @property
+    def file_type(self):
+        return get_default_file_extension(self.mime_type)
+
+    @property
+    def source_path(self) -> Path:
+        if self.filename:
+            fname = str(self.filename)
+        return (settings.ORIGINALS_DIR / Path(fname)).resolve()
+
+
+def add_number_of_pages_to_pages_count(apps, schema_editor):
+    documentModel = apps.get_model("documents", "Document")
+
+    if not documentModel.objects.all().exists():
+        return
+
+    for doc in documentModel.objects.filter(mime_type="application/pdf"):
+        document = Document(doc)
+
+        print(
+            "    {} {} {}".format(
+                colourise("*", fg="green"),
+                colourise("Calculating number of pages for", fg="white"),
+                colourise(document.filename, fg="cyan"),
+            ),
+        )
+
+        pdf = pikepdf.open(document.source_path)
+
+        if pdf.pages is not None:
+            doc.pages_count = len(pdf.pages)
+            doc.save()
+
+
+def remove_number_of_pages_to_pages_count(apps, schema_editor):
```

Comment:
Is there a purpose to setting these to 0 in a reverse migration? The field would get removed anyway...

Reply:
> Is there a purpose to setting these to 0 in a reverse migration? The field would get removed anyway...

I saw that in previous migrations it was also done this way, so i wanted to follow the same principle.
I don't think it's actually necessary, but Pytest requires it, otherwise it throws an error.
Otherwise, should i create a function that do nothing ?

Reply:
What happens if the reverse if `migrations.RunPython.noop`?

Reply:
Using `migrations.RunPython.noop` instead of `remove_number_of_pages_to_pages_count` doesn't seem to cause any issues with Pytest or during the migration when I run the test. I do think it's indeed the best solution.